### PR TITLE
Gated video fix for issue #3349

### PR DESCRIPTION
**Pull request by MarkLL**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #3349, #3177  
| BC breaks? | 
| Deprecations? | 

#### Description:
This PR fixes an issue where Gated videos did not render correctly if included in Dynamic Content. I have also attempted to fix another outstanding issue were other videos were rendered incorrectly.

Basically if a gated video is detected in the response for Dynamic Content, `MauticJS.initGatedVideo()` is called again to detect the new DOM content.

#### Steps to test this PR:
1. Run the steps outlined in #3349 and verify you see the behavior.
2. Install this PR and test again. This time you should see the video rendered correctly.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. See #3349

### CODE REVIEW

**Comment by mqueme** [comment]
gated videos continued to work for me normally after this PR. I cannot test it with the mediaElements .js though

**Comment by escopecz** [comment]
Gated video in a DWC slot worked for me on staging as well as on this PR, so no reason why not to merge it 👍

**Comment by alanhartless** [comment]
This looks to include code from #3348. If it's to be merged out of sync with that, the code needs to be removed from this PR.

**Comment by escopecz** [comment]
Since I approved #3348 I'm giving a green light to this PR too (which I had tested before). I merged in latest staging to kick the Travis tests again.

**Comment by MarkLL** [comment]
thanks for assisting with the merges 👍

**Comment by leonardoborlot** [comment]
Please guys, these resource is very good!

**Comment by virlatinus** [comment]
Tested this PR and it works!
